httpcache.go

```diff
@@ -227,9 +229,22 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 				resp.Header.Set(fakeHeader, reqValue)
 			}
 		}
-		respBytes, err := httputil.DumpResponse(resp, true)
-		if err == nil {
-			t.Cache.Set(cacheKey, respBytes)
+		switch req.Method {
+		case "GET":
+			// Delay caching until EOF is read.
+			resp.Body = newCachedReadCloser(resp.Body, func(r io.ReadCloser) {
```

Comment:
I suspect it would be more clear what's going on here if you assigned the 2nd parameter to an `onEOF` variable, and then passed that to `newCachedReadCloser`. As is, one has to be intimately familiar with the `newCachedReadCloser` signature to be able to see what's happening here. So:

```Go
onEOF := func(r io.ReadCloser) {
	response := *resp
	response.Body = r
	respBytes, err := httputil.DumpResponse(&response, true)
	if err == nil {
		t.Cache.Set(cacheKey, respBytes)
	}
}
resp.Body = newCachedReadCloser(resp.Body, onEOF)